### PR TITLE
add metrics for ldClient initialization

### PR DESCRIPTION
**Pull request by adrianliu0815**
This is an effort to have better observability about ldClient startup. 
Added 3 metrics in this PR:
1. success count
2. failed  count
3. initialization latency

### CODE REVIEW

File: wisp-launchdarkly/src/main/kotlin/wisp/launchdarkly/LaunchDarklyClientMetrics.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    .register(meterRegistry)
+
+  internal val failedCount: Counter = Counter.builder(FAILED_COUNTER_NAME)
+    .description("count of ld client initialization failures")
+    .register(meterRegistry)
+
+  companion object {
+    const val HISTO_DURATION_NAME = "histo_ld_initialization_duration_ms"
+    const val SUCCESS_COUNTER_NAME = "ld_initialization_success"
+    const val FAILED_COUNTER_NAME = "ld_initialization_failed"
```
Comment (chris-ryan-square) [inline wisp-launchdarkly/src/main/kotlin/wisp/launchdarkly/LaunchDarklyClientMetrics.kt:28]: Should these have "_total" for compatibility with prometheus?

Comment (adrianliu0815) [reply wisp-launchdarkly/src/main/kotlin/wisp/launchdarkly/LaunchDarklyClientMetrics.kt]: added.

File: wisp-launchdarkly/src/main/kotlin/wisp/launchdarkly/LaunchDarklyFeatureFlags.kt [hunk 2]
```diff
@@ -8,36 +8,50 @@ import com.launchdarkly.sdk.server.interfaces.LDClientInterface
...
+            }
 
-        if (attempts == 0 && !ldClient.isInitialized) {
-            throw Exception("LaunchDarkly did not initialize in 30 seconds")
+            // ldClient successfully initialized
+            launchDarklyClientMetrics.successCount.increment()
         }
+        launchDarklyClientMetrics
+            .initializationDuration(meterRegistry)
+            .record(timedResult.toDouble())
```
Comment (chris-ryan-square) [inline wisp-launchdarkly/src/main/kotlin/wisp/launchdarkly/LaunchDarklyFeatureFlags.kt:54]: Do you want this to capture the time spent for unsuccessful initializations, the exception above will avoid this?  Not sure if it's desired or not.

**Review by chris-ryan-square — APPROVED** [review]

**Comment by chris-ryan-square** [comment]
You will need to run the apiDump gradle task for these changes.

File: wisp-launchdarkly/src/main/kotlin/wisp/launchdarkly/LaunchDarklyFeatureFlags.kt (hunk 2 above)
Comment (adrianliu0815) [reply wisp-launchdarkly/src/main/kotlin/wisp/launchdarkly/LaunchDarklyFeatureFlags.kt:54]: unsuccessful initializations would always be ~30s and I will just use failure count to capture that.

File: wisp-launchdarkly/api/wisp-launchdarkly.api [hunk 3]
```diff
@@ -3,6 +3,17 @@ public final class wisp/launchdarkly/LaunchDarklyClient {
 	public final fun createLaunchDarklyClient (Lwisp/launchdarkly/LaunchDarklyConfig;Lwisp/security/ssl/SslLoader;Lwisp/security/ssl/SslContextFactory;Lwisp/resources/ResourceLoader;)Lcom/launchdarkly/sdk/server/interfaces/LDClientInterface;
 }
 
+public final class wisp/launchdarkly/LaunchDarklyClientMetrics {
```
Comment (adrianliu0815) [inline wisp-launchdarkly/api/wisp-launchdarkly.api:6]: generated by 
./bin/gradle :wisp-launchdarkly:apiDump

**Pull request merged**